lib/v2/chinawriter/index.js

```diff
@@ -0,0 +1,88 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const { id } = ctx.params;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 40;
+
+    const rootUrl = 'http://www.chinawriter.com.cn';
+    const currentUrl = new URL(id ?? '', rootUrl).href;
+
+    const { data: response } = await got(currentUrl);
+
+    const $ = cheerio.load(response);
+
+    let items = ($('main div.inner').find('a').length === 0 ? $('body') : $('main div.inner'))
+        .find('a')
+        .toArray()
+        .filter((item) => /\/\d{4}\/\d{4}\/\w+-\w+\.html/.test($(item).prop('href')))
+        .slice(0, limit)
+        .map((item) => {
+            item = $(item);
+
+            const link = item.prop('href');
+
+            return {
+                title: item.text(),
+                link: /^http/.test(link) ? link : new URL(item.prop('href'), rootUrl).href,
+            };
+        });
+
+    items = await Promise.all(
+        items.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                try {
+                    const { data: detailResponse } = await got(item.link);
+
+                    const content = cheerio.load(detailResponse);
+
+                    content('div.end_shared').remove();
+
+                    const info = content('div.end_info').text().trim();
+
+                    item.title = content('#newstit').text() || content('h6.end_tit').text();
+                    item.description = content('div.end_article').html();
+                    item.author = info ? info.match(/\|(.*)\d{4}/)[1].trim() : '';
+                    item.category = [
+                        ...new Set(
+                            [
+                                ...content('div.location a.clink')
+                                    .slice(1)
+                                    .toArray()
+                                    .map((c) => content(c).text()),
+                                info
+                                    ? info
+                                          .match(/^(.*)\|/)[1]
+                                          .replace(/来源：/g, '')
+                                          .trim()
+                                    : undefined,
+                            ].filter(Boolean)
+                        ),
+                    ];
+                    item.pubDate = content('div.end_info em').text() ? timezone(parseDate(content('div.end_info em').text(), 'YYYY年MM月DD日HH:mm'), +8) : parseDate(content('meta[name="description"]').prop('content'));
```

Comment:
```suggestion
                    item.pubDate = content('div.end_info em').text() ? timezone(parseDate(content('div.end_info em').text(), 'YYYY年MM月DD日HH:mm'), +8) : parseDate(content('meta[name="publishdate"]').prop('content'));
```
https://github.com/DIYgod/RSSHub/pull/13271#discussion_r1323445876

So `http://www.chinawriter.com.cn/n1/2023/0303/c404928-32635716.html` from `/chinawriter/403978` can have proper pubDate.